simulation/traffic_simulator/src/entity/entity_manager.cpp

```diff
@@ -253,14 +253,24 @@ auto EntityManager::getLongitudinalDistance(
   bool include_adjacent_lanelet, bool include_opposite_direction) -> std::optional<double>
 {
   if (!include_adjacent_lanelet) {
-    auto forward_distance = hdmap_utils_ptr_->getLongitudinalDistance(
-      static_cast<LaneletPose>(from), static_cast<LaneletPose>(to));
+    auto to_canonicalized = static_cast<LaneletPose>(to);
+    if (to.hasAlternativeLaneletPose()) {
+      to_canonicalized = to.getAlternativeLaneletPoseBaseOnShortestRouteFrom(
+        static_cast<LaneletPose>(from), hdmap_utils_ptr_);
+    }
+
+    auto forward_distance =
+      hdmap_utils_ptr_->getLongitudinalDistance(static_cast<LaneletPose>(from), to_canonicalized);
 
-    auto backward_distance = hdmap_utils_ptr_->getLongitudinalDistance(
-      static_cast<LaneletPose>(to), static_cast<LaneletPose>(from));
+    auto backward_distance =
+      hdmap_utils_ptr_->getLongitudinalDistance(to_canonicalized, static_cast<LaneletPose>(from));
 
     if (forward_distance && backward_distance) {
```

Comment:
```suggestion
    if (const auto forward_distance = hdmap_utils_ptr_->getLongitudinalDistance(static_cast<LaneletPose>(from), to_canonicalized) && const auto backward_distance = hdmap_utils_ptr_->getLongitudinalDistance(to_canonicalized, static_cast<LaneletPose>(from))) {
```

Reply:
I'm not able to apply this code change suggestion because a few lines lower the `forward_distance` variable and the `backward_distance` variable is used in `else if ()` statement.

Reply:
I see. Thanks.